feat(floating-label): add feature targeting for styles

crisbeto opened the pull request:
Adds support for feature targeting to the `mdc-floating-label` styles.

Relates to #4227.

crisbeto commented on test/scss/_feature-targeting-test.scss:
```diff
@@ -159,6 +160,15 @@
     @include mdc-elevation-overlay-opacity(99%, $query: $query);
 
+    // Floating label
+    @include mdc-floating-label-core-styles($query: $query);
```
Note that the `test:feature-targeting` check doesn't seem to be running on the CI because there are failures in the files that's in master. I've added my changes here for completeness.

crisbeto commented on packages/mdc-floating-label/_mixins.scss:
```diff
@@ -21,60 +21,148 @@
+}
+
+@mixin mdc-floating-label-shake-keyframes($modifier, $positionY, $positionX: 0%, $scale: .75,
+$query: mdc-feature-all()) {
```
The indentation here is weird because the top line was over the limit so I had to break it up, but stylelint starts complaining if I add any indentation to it.

mmalerba commented on packages/mdc-floating-label/mdc-floating-label.scss:
```diff
@@ -20,60 +20,9 @@
-  }
-
-  @include mdc-floating-label-shake-keyframes(standard, $mdc-floating-label-position-y);
+  @include mdc-floating-label-core-styles($query: mdc-feature-all());
```
nit: this is the default `$query`, can just not pass any params

crisbeto commented:
I've addressed the feedback @mmalerba and I've attached the before/after files. There are a few lines of difference, but they're within the same rule and shouldn't have an effect at runtime. I moved these lines down so that they could fit into an existing feature target.

[before.css.txt](https://github.com/material-components/material-components-web/files/3933489/before.css.txt)
[after.css.txt](https://github.com/material-components/material-components-web/files/3933491/after.css.txt)

abhiomkar commented on packages/mdc-floating-label/_mixins.scss:
```diff
@@ -21,60 +21,149 @@
-}
+  // postcss-bem-linter: define floating-label
+  .mdc-floating-label {
+    @include mdc-typography(subtitle1, $query: $query);
```
nit: Please use `$exclude-props` to exclude `line-height` in typography to avoid duplication with below CSS override.

BUILD will throw an error otherwise. :)

`/* @alternative */` comment before CSS override would also fix this.

crisbeto replied:
Fixed.
